### PR TITLE
Formik discount modal

### PR DESCRIPTION
**Pull request by RomaKushnir**

### CODE REVIEW

**Comment by RomaKushnir** [comment]
don't know is it ok, I've created class "formDisplayNone" on our form so after request is successful it was possible to show smal modal of confirmation

File: src/pages/Discounts/components/CreateDiscount/CreateDiscount.js [hunk 1]
```diff
@@ -41,8 +27,47 @@ function CreateDiscount({
...
+    : null;
+
+  const initialCategories = discount ? {
+    value: discount.category.id,
+    label: discount.category.title
+  }
+    : null;
 
-  console.log('createDiscountStatus', createDiscountStatus);
+  const initialLocations = discount
```
Comment (artemtool) [inline src/pages/Discounts/components/CreateDiscount/CreateDiscount.js:45]: Use "useMemo" here not to remap on each rerender.

File: src/pages/Discounts/components/CreateDiscount/CreateDiscount.js [hunk 2]
```diff
@@ -41,8 +27,47 @@ function CreateDiscount({
...
+  const discountRequest = {
+    title: discount ? discount.title : '',
+    imageUrl: discount ? discount.imageUrl : '',
+    description: discount ? discount.description : '',
+    shortDescription: discount ? discount.shortDescription : '',
+    flatAmount: discount ? discount.flatAmount : '',
+    percentage: discount ? discount.percentage : '',
+    startDate: discount ? new Date(discount.startDate) : new Date(Date.now()),
+    expirationDate: discount ? new Date(discount.expirationDate) : null,
+    locationIds: discount ? discount.locations.map((el) => (el.id)) : [],
```
Comment (artemtool) [inline src/pages/Discounts/components/CreateDiscount/CreateDiscount.js:63]: Extract this to variable and use "useMemo" not to remap on each rerender.

File: src/store/sagas/discountsSagas.js [hunk 3]
```diff
@@ -23,13 +23,18 @@ export function* getDiscounts() {
...
-
-    yield put(actions.discountsActions.createDiscountSuccess(response.data));
+    // console.log('SAGA createDiscount');
+    if (!id) {
+      response = yield call(api.discounts.createDiscount, data);
+      yield put(actions.discountsActions.createDiscountSuccess(response.data));
+    } else {
+      response = yield call(api.discounts.updateDiscount, payload);
+      yield put(actions.discountsActions.createDiscountSuccess(response.data));
+    }
```
Comment (artemtool) [inline src/store/sagas/discountsSagas.js:37]: You can put this action after if-else block and write it once.

File: src/pages/Discounts/components/CreateDiscount/CreateDiscount.js [hunk 4]
```diff
@@ -100,7 +139,8 @@ function CreateDiscount({
       && <div className = {styles.loadingContainer}>
         <CircularProgress />
       </div>}
-      <form>
+      <form className={createDiscountStatus.loading === false && createDiscountStatus.success
+        && styles.formDisplayNone}>
```
Comment (artemtool) [inline src/pages/Discounts/components/CreateDiscount/CreateDiscount.js:143]: Extract this expression to variable and use variable here.

**Review by artemtool — CHANGES_REQUESTED** [review]
Good job!
However small changes required.

**Review by artemtool — APPROVED** [review]

**Pull request merged**